client/solve.go

```diff
@@ -79,28 +80,30 @@ func (c *Client) Solve(ctx context.Context, def *llb.Definition, opt SolveOpt, s
 
 	switch opt.Exporter {
 	case ExporterLocal:
-		outputDir, ok := opt.ExporterAttrs[exporterLocalOutputDir]
-		if !ok {
-			return errors.Errorf("output directory is required for local exporter")
+		if opt.ExporterOutputFile != nil {
+			logrus.Warnf("output file writer is ignored for local exporter")
+		}
+		// it is ok to have empty output dir (just ignored)
+		// FIXME(AkihiroSuda): maybe disallow empty output dir? (breaks integration tests)
+		if opt.ExporterOutputDir != "" {
+			s.Allow(filesync.NewFSSyncTarget(opt.ExporterOutputDir))
 		}
-		s.Allow(filesync.NewFSSyncTarget(outputDir))
 	case ExporterOCI, ExporterDocker:
-		outputFile, ok := opt.ExporterAttrs[exporterOCIDestination]
-		if ok {
-			fi, err := os.Stat(outputFile)
-			if err != nil && !os.IsNotExist(err) {
-				return errors.Wrapf(err, "invlid destination file: %s", outputFile)
-			}
-			if err == nil && fi.IsDir() {
-				return errors.Errorf("destination file is a directory")
-			}
-		} else {
-			if _, err := console.ConsoleFromFile(os.Stdout); err == nil {
-				return errors.Errorf("output file is required for %s exporter. refusing to write to console", opt.Exporter)
-			}
-			outputFile = ""
+		if opt.ExporterOutputDir != "" {
+			logrus.Warnf("output directory %s is ignored for %s exporter", opt.ExporterOutputDir, opt.Exporter)
+		}
+		// it is ok to have empty output file (just ignored)
+		// FIXME(AkihiroSuda): maybe disallow empty output file? (breaks integration tests)
```

Comment:
I think we should not allow it

Reply:
ok, but let me work on this separately after I get CI green

https://github.com/moby/buildkit/issues/327